src/main/java/org/fogbowcloud/manager/core/SharedOrderHolders.java

```diff
@@ -20,13 +23,30 @@
     private SynchronizedDoublyLinkedList closedOrders;
 
     private SharedOrderHolders() {
-        this.activeOrdersMap = new ConcurrentHashMap<String, Order>();
-        this.openOrders = new SynchronizedDoublyLinkedList();
-        this.spawningOrders = new SynchronizedDoublyLinkedList();
-        this.failedOrders = new SynchronizedDoublyLinkedList();
-        this.fulfilledOrders = new SynchronizedDoublyLinkedList();
-        this.pendingOrders = new SynchronizedDoublyLinkedList();
-        this.closedOrders = new SynchronizedDoublyLinkedList();
+        DatabaseManager databaseManager = DatabaseManager.getInstance();
+
+        this.activeOrdersMap = new ConcurrentHashMap<>();
+
+        this.openOrders = databaseManager.readActiveOrders(OrderState.OPEN);
+        addOrdersToMap(this.openOrders, this.activeOrdersMap);
+        this.spawningOrders = databaseManager.readActiveOrders(OrderState.SPAWNING);
+        addOrdersToMap(this.spawningOrders, this.activeOrdersMap);
+        this.failedOrders = databaseManager.readActiveOrders(OrderState.FAILED);
+        addOrdersToMap(this.failedOrders, this.activeOrdersMap);
+        this.fulfilledOrders = databaseManager.readActiveOrders(OrderState.FULFILLED);
+        addOrdersToMap(this.fulfilledOrders, this.activeOrdersMap);
+        this.pendingOrders = databaseManager.readActiveOrders(OrderState.PENDING);
+        addOrdersToMap(this.pendingOrders, this.activeOrdersMap);
+        this.closedOrders = databaseManager.readActiveOrders(OrderState.CLOSED);
+        addOrdersToMap(this.closedOrders, this.activeOrdersMap);
```

Comment:
what is the idiom of using this List? Are one supposed to reset it whenever it knows the end was reached?

Reply:
I don't got it.

Reply:
What Thiago means is that once you have iterated in the list until its end, you should reset the pointer, so that the next thread that iterates in the list will start from the beginning. This is just a performance improvement.

Reply:
Done!

Reply:
ok